File: src/main/code/com/melonbase/microquark/rest/HelloResource.java
package com.melonbase.microquark.rest;

import com.melonbase.microquark.service.HelloService;

import javax.inject.Inject;
import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;

@Path("/hello")
public class HelloResource {

  @Inject
  HelloService helloService;

  @GET
  @Produces(MediaType.TEXT_PLAIN)
  public String getHello() {
    return helloService.getHello();
  }

  @POST
  @Consumes(MediaType.TEXT_PLAIN)
  public void setHello(String msg) {
    helloService.setHello(msg);
  }
}

File: src/main/code/com/melonbase/microquark/rest/HelloResource.kt
package com.melonbase.microquark.rest

import com.melonbase.microquark.service.HelloService
import javax.inject.Inject
import javax.ws.rs.*
import javax.ws.rs.core.MediaType

@Path("hello")
class HelloResource @Inject constructor(val helloService: HelloService) {

  @GET
  @Produces(MediaType.TEXT_PLAIN)
  fun getHello(): String? {
    return helloService.getHello()
  }

  @POST
  @Consumes(MediaType.TEXT_PLAIN)
  fun setHello(msg: String) {
    helloService.setHello(msg)
  }
}